tests/acceptance/features/bootstrap/WebDav.php

```diff
@@ -4463,6 +4483,85 @@ public function userHasDeletedEverythingInFolder(string $user, string $folder):v
 		$this->userDeletesEverythingInFolder($user, $folder, true);
 	}
 
+	/**
+	 * @When user :user tries to download the preview of nonexistent file :path with width :width and height :height using the WebDAV API
+	 *
+	 * @param string $user
+	 * @param string $path
+	 * @param string $width
+	 * @param string $height
+	 *
+	 * @return void
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function userTriesToDownloadThePreviewOfNonexistentFileWithWidthAndHeightUsingTheWebdavApi(string $user, string $path, string $width, string $height):void {
+		$urlParameter = [
+			'x' => $width,
+			'y' => $height,
+			'forceIcon' => '0',
+			'preview' => '1'
+		];
+
+		$user = $this->getActualUsername($user);
+		$davVersion = $this->getDavPathVersion();
+
+		if ($davVersion === WebDavHelper::DAV_VERSION_SPACES) {
+			// generate fake fileId
+			$fakeFileId = WebDavHelper::generateUUIDv4();
+			$fileId = $fakeFileId . '!' . $fakeFileId;
+			$encodedFileId = urlencode(strstr($fileId, '!', true));
+
+			// generate 32-character random etag string
+			$eTag = bin2hex(random_bytes(16));
+
+			$url = $this->getBaseUrl() . '/remote.php/dav/spaces/' . $encodedFileId . '/';
+			$urlParameter = [
+				'scalingup' => 0,
+				'preview' => '1',
+				'a' => '1',
+				'c' => $eTag,
+				'x' => $width,
+				'y' => $height
+			];
+		} elseif ($davVersion === WebDavHelper::DAV_VERSION_NEW) {
+			$url = $this->getBaseUrl() . '/remote.php/dav/files/' . $user . '/';
+		} else {
+			$url = $this->getBaseUrl() . '/remote.php/webdav/';
+		}
+
+		$urlParameter = \http_build_query($urlParameter, '', '&');
+		$path .= '?' . $urlParameter;
+		$fullUrl = WebDavHelper::sanitizeUrl($url . $path);
+
+		$this->response = HttpRequestHelper::sendRequest(
+			$fullUrl,
+			'',
+			'GET',
+			$user,
+			$this->getPasswordForUser($user)
+		);
+	}
```

Comment:
instead of creating new step cannot we reuse the old one and adjust the `dav-verison` code there?

Reply:
this preview one is a little bit complex to adjust in old code. As you can see  we dont need space id here like other space dav version